### PR TITLE
Db/update test data input

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

[#102 ](https://github.com/OpsLevel/team-platform/issues/102)

## Changelog

Add `NewTestRequest()` constructor that validates requests, variables, and responses before returning a `TestRequest` struct that represents the data under test. Update `actions_test.go` to use these updates.
- [X] List your changes here
- [ ] Make a `changie` entry <--- test only updates. Not needed for this PR.

## Tophatting

`task test` demonstrates all tests pass

### CODE REVIEW

File: clientGQL_test.go [hunk 1]
```diff
@@ -83,6 +140,11 @@ func RegisterPaginatedEndpoint(t *testing.T, endpoint string, requests ...TestRe
 	return autopilot.Server.URL + url
 }
 
+func TmpBetterTestClient(t *testing.T, endpoint string, testRequest TestRequest) *ol.Client {
+	oldStyleRequest := fmt.Sprintf(`{%s, %s}`, testRequest.Request, testRequest.Variables)
```
Comment (rocktavious) [inline clientGQL_test.go:144]: we shouldn't do this - this is likely making things WAY more complicated.

We should make the new `TheBestTestClient` which natively works with `[]TestRequest`

I have a pretty clear picture in my head of how that will work - do you want me to mock it up for you?

Comment (davidbloss) [reply clientGQL_test.go:144]: To be updated per offline conversation

File: clientGQL_test.go [hunk 2]
```diff
@@ -66,33 +76,159 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
...
+	t.Variables = fmt.Sprintf("%s %s", varPrefix, jsonFormattedVariableObject)
+}
+
+func (t *TestRequest) ParseResponse(rawResponse string) {
+	parsedResponse, err := t.templater.ParseTemplatedString(rawResponse)
+	if err != nil {
+		panic(err)
+	}
+	if !t.IsValidJson(parsedResponse) {
+		panic(fmt.Errorf("invalid json: %s", parsedResponse))
```
Comment (taimoor-at-opslevel) [inline clientGQL_test.go:202]:
```suggestion
		panic(fmt.Errorf("test response could not be JSON formatted: %s", parsedResponse))
```

Just for consistency

**Review by taimoor-at-opslevel — APPROVED** [review]

**Review by taimoor-at-opslevel — APPROVED** [review]

File: alert_source_test.go [hunk 3]
```diff
@@ -9,30 +9,13 @@ import (
...
-			}
-		}
-	}
-}}`
-	client := ABetterTestClient(t, "alert_source/create", request, response)
+	testRequest := NewTestRequest(
+		`"query": "mutation AlertSourceServiceCreate($input:AlertSourceServiceCreateInput!){alertSourceServiceCreate(input: $input){alertSourceService{alertSource{name,description,id,type,externalId,integration{id,name,type},url},id,service{id,aliases},status},errors{message,path}}}"`,
+		`"variables": {"input": { "alertSourceExternalIdentifier": { "externalId": "QWERTY", "type": "datadog" }, "service": { "alias": "example" }}}`,
+		`{"data": { "alertSourceServiceCreate": { "alertSourceService": { "service": { "aliases": ["example"] }}}}}`,
+	)
```
Comment (rocktavious) [inline alert_source_test.go:16]:
```suggestion
	testRequest := NewTestRequest(
		`mutation AlertSourceServiceCreate($input:AlertSourceServiceCreateInput!){alertSourceServiceCreate(input: $input){alertSourceService{alertSource{name,description,id,type,externalId,integration{id,name,type},url},id,service{id,aliases},status},errors{message,path}}}`,
		`{"input": { "alertSourceExternalIdentifier": { "externalId": "QWERTY", "type": "datadog" }, "service": { "alias": "example" }}`,
		`{"data": { "alertSourceServiceCreate": { "alertSourceService": { "service": { "aliases": ["example"] }}}}}`,
	)
```

File: clientGQL_test.go [hunk 4]
```diff
@@ -66,36 +71,94 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
...
+	return strings.TrimSpace(data.String())
+}
+
 type TestRequest struct {
 	Request   string
 	Variables string
 	Response  string
 }
 
+func (t *TestRequest) RequestWithVariables() string {
```
Comment (rocktavious) [inline clientGQL_test.go:138]: @davidbloss can we log a follow up ticket to get rid of this.  I understand why its like this now but i'd love to kill this in a follow up.

File: clientGQL_test.go [hunk 5]
```diff
@@ -66,36 +71,94 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
...
+		Variables: templater.ParseTemplatedString(variables),
+		Response:  templater.ParseTemplatedString(response),
+	}
+	if !IsValidJson(testRequest.Variables) {
+		panic(fmt.Errorf("testRequest Variables is not valid json: %s", testRequest.Variables))
+	}
+	if !IsValidJson(testRequest.Response) {
+		panic(fmt.Errorf("testRequest Response is not json: %s", testRequest.Response))
+	}
+	return testRequest
```
Comment (rocktavious) [inline clientGQL_test.go:87]: 😭 - so beautiful it makes me cry

File: clientGQL_test.go [hunk 6]
```diff
@@ -66,36 +71,94 @@ func ATestClient(t *testing.T, endpoint string) *ol.Client {
...
+}
+
+func NewTestDataTemplater(templateDirs ...string) *TestDataTemplater {
+	var templateFiles []string
+	for _, dir := range []string{"./testdata/templates"} {
+		err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
```
Comment (rocktavious) [inline clientGQL_test.go:95]: the signature of this function doesn't return an error yet we are here and the caller is hand handling an error.  Maybe we should just panic to the end user knows theres a broken template?

**Review by rocktavious — APPROVED** [review]
LGTM

**Pull request merged**